app/scripts/components/exploration/components/timeline/timeline-controls.tsx

```diff
@@ -255,26 +304,137 @@ const getCalendarView = (timeDensity: TimeDensity): View => {
 };
 
 export function TimelineControls(props: TimelineControlsProps) {
-  const { xScaled, width, outOfViewHeads, onZoom, timeDensity, timelineLabelsFormat, minMaxTemporalExtent } = props;
+  const {
+    xScaled,
+    width,
+    outOfViewHeads,
+    onZoom,
+    timeDensity,
+    timelineLabelsFormat,
+    minMaxTemporalExtent
+  } = props;
 
   const [selectedDay, setSelectedDay] = useAtom(selectedDateAtom);
+
   const [selectedCompareDay, setSelectedCompareDay] = useAtom(
     selectedCompareDateAtom
   );
   const [selectedInterval, setSelectedInterval] = useAtom(selectedIntervalAtom);
   const { features } = useAois();
 
   // Scale to use when there are no datasets with data (loading or error)
-  const initialScale = useMemo(() => getInitialScale(width) ,[width]);
+  const initialScale = useMemo(() => getInitialScale(width), [width]);
 
-  const calendarView = useMemo(() => getCalendarView(timeDensity), [timeDensity]);
+  const calendarView = useMemo(
+    () => getCalendarView(timeDensity),
+    [timeDensity]
+  );
+  //Center to selected point
+  const { onTOIZoom } = useOnTOIZoom();
+  const timelineWidth = useAtomValue(timelineWidthAtom);
+  const { k: currentZoomTransformRatio } = useAtomValue(zoomTransformAtom);
+  const { main } = useScales();
+  const visualBufferSizing = 0.9;
+  const startPoint = 0;
+
+  const calculateNewTOIZoom = (
+    dateStart,
+    dateEnd,
+    widthToFit
+  ): { zTransform: number; panPosition: number } => {
+    const zTransform = widthToFit / (dateEnd - dateStart);
+    const panPosition = startPoint - zTransform * dateStart;
+    return { zTransform, panPosition };
+  };
+
+  const centerTimelineOnSelections = useCallback(
+    (newDate) => {
```

Comment:
Can we add a type for the `newDate` param? Also the params for `calculateNewTOIZoom` (dateStart, dateEnd, widthToFit)?

Reply:
Of course!

Reply:
I added an additional date fallback, in the edge case that we do not receive a date from the on confirm function and we do not set a date in state.

Reply:
Ignore the above comment, it introduced an odd error that crashed the functionality when executed.